spec/praxis-mapper/connection_factories/sequel_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'spec_helper'
+
+describe Praxis::Mapper::ConnectionFactories::Sequel do
+
+  let(:database) { Sequel.mock}
+  let(:thread) { Thread.current }
+  let(:connection_manager) { double('Praxis::Mapper::ConnectionManager', thread: thread) }
+
+  subject(:factory) { described_class.new(connection:database) }
+
+  context 'checkout' do
+    it 'returns the connection' do
+      factory.checkout(connection_manager).should be database
+    end
+
+    it 'allocates a connection to the thread of the manager' do
+      database.pool.allocated.should_not have_key(thread)
+      factory.checkout(connection_manager)
+      database.pool.allocated.should have_key(thread)
+    end
+
+  end
+
+  context 'release' do
+    before do
+      factory.checkout(connection_manager)
+    end
+
+    it 'releases the connection' do
+      database.pool.allocated.should have_key(thread)
+      factory.release(connection_manager, database)
+      database.pool.allocated.should_not have_key(thread)
+    end
+  end
+
+  context 'across multiple threads' do
+    before do
+      factory.checkout(connection_manager)
+    end
+
+    it 'works somehow' do
```

Comment:
should change the test name

Also, shouldn't we somehow test that thread1 gets an allocated connection first, then spawns thread2 and we can have thread 2 check that the pool has not only his connection allocated, but another one for thread1?